e107_plugins/forum/forum_class.php

```diff
@@ -2527,27 +2527,24 @@ function threadDelete($threadId, $updateForumLastpost = true)
 	 * Delete a Post
 	 * @param $postId integer
 	 * @param $updateCounts boolean
-	 * 
+	 * @return "null" if this post does not exist, "true" if post could deleted, otherwise "false"
 	 */
 	function postDelete($postId, $updateCounts = true)
 	{
 		$postId 	= (int)$postId;
-		$e107 		= e107::getInstance();		
+
 		$sql 		= e107::getDb();
 		$deleted 	= false;
 
 		$postInfo   = $sql->retrieve('forum_post', '*', 'post_id = '.$postId);
-		//if(!$sql->select('forum_post', '*', 'post_id = '.$postId))
+
 		if(!is_array($postInfo) || empty($postInfo))
 		{
-			echo 'NOT FOUND!'; return;
+			return null;
```

Comment:
Changed the return here to "null" and adapted the description.

Reply:
:+1: